resources/js/components/attribute/Attribute.vue

```diff
@@ -0,0 +1,428 @@
+
+    export default {
+        components: {
+            'string-attribute': StringAttr,
+            'stringfield-attribute': Stringfield,
+            'richtext-attribute': Richtext,
+            'integer-attribute': IntegerAttr,
+            'float-attribute': FloatAttr,
+            'bool-attribute': Bool,
+            'percentage-attribute': Percentage,
+            'serial-attribute': Serial,
+            'dimension-attribute': Dimension,
+            'epoch-attribute': Epoch,
+            'list-attribute': List,
+            'tabular-attribute': Tabular,
+            'iconclass-attribute': Iconclass,
+            'rism-attribute': RISM,
+            'geography-attribute': Geography,
+            'entity-attribute': Entity,
+            'date-attribute': DateAttr,
+            'daterange-attribute': DaterangeAttr,
+            'singlechoice-attribute': SingleChoice,
+            'multichoice-attribute': MultiChoice,
+            'userlist-attribute': UserList,
+            'url-attribute': Url,
+            'sql-attribute': SqlAttr,
+            'system-separator-attribute': SystemSeparator,
+            'default-attribute': DefaultAttr,
+        },
+        props: {
+            data: {
+                type: Object,
+                required: true,
+            },
+            valueWrapper: {
+                type: Object,
+                required: false,
+                default: _ => new Object(),
+            },
+            disabled: {
+                type: Boolean,
+                required: false,
+                default: false,
+            },
+            reactTo: {
+                type: Number,
+                required: false,
+                default: -1,
+            },
+            hideLinks: {
+                type: Boolean,
+                required: false,
+                default: false,
+            },
+            preview: {
+                type: Boolean,
+                required: false,
+                default: false,
+            },
+            previewData: {
+                type: Object,
+                required: false,
+                default: _ => new Object(),
+            },
+        },
+        emits: ['change', 'update-selection'],
+        setup(props, context) {
+            const {
+                data,
+                valueWrapper,
+                disabled,
+                reactTo,
+                hideLinks,
+                preview,
+                previewData,
+            } = toRefs(props);
+            // FETCH
+
+            const attrRef = ref({});
+            const state = reactive({
+                type: computed(_ => data.value.datatype),
+                disabled: computed(_ => data.value.isDisabled || disabled.value),
+                value: computed(_ => getValueOrDefault()),
+                // TODO check for selection need?
+                selection: computed(_ => getAttributeSelection(data.value.id) || {}),
```

Comment:
This is a problem, I think this should return an array as default => ... | [ ]
Throws error when creating table attribute.

Reply:
Yeah, selections should be Arrays 👍